exonum/src/blockchain/transaction.rs

```diff
@@ -343,6 +357,54 @@ mod tests {
         static ref EXECUTION_STATUS: Mutex<ExecutionResult> = Mutex::new(Ok(()));
     }
 
+    #[test]
+    fn execution_error_new() {
+        let codes = [0, 1, 100, 255];
+
+        for code in &codes {
+            let error = ExecutionError::new(*code);
+            assert_eq!(*code, error.code);
+            assert_eq!(None, error.description);
+        }
+    }
+
+    #[test]
+    fn execution_error_with_description() {
+        let values = [
+            (0, "".to_owned()),
+            (1, "test".to_owned()),
+            (100, "errro".to_owned()),
+            (255, "hello".to_owned()),
+        ];
+
+        for value in &values {
+            let error = ExecutionError::with_description(value.0, value.1.clone());
+            assert_eq!(value.0, error.code);
+            assert_eq!(value.1, error.description.unwrap());
+        }
+    }
+
+    #[test]
+    fn transaction_error_new() {
+        let values = [
+            (TransactionErrorType::Panic, None),
+            (TransactionErrorType::Panic, Some("panic".to_owned())),
+            (TransactionErrorType::Code(0), None),
+            (TransactionErrorType::Code(1), Some("".to_owned())),
+            (TransactionErrorType::Code(100), None),
+            (
+                TransactionErrorType::Code(255),
+                Some("error description".to_owned()),
+            ),
+        ];
+
+        for value in &values {
```

Comment:
Would it make sense to override the test as follows?

```rust
let values = [
    (TransactionErrorType::Panic, None),
    (TransactionErrorType::Panic, Some("panic")),
    (TransactionErrorType::Code(0), None),
    (TransactionErrorType::Code(1), Some("")),
    (TransactionErrorType::Code(100), None),
    (
        TransactionErrorType::Code(255),
        Some("error description"),
    ),
];

for value in values.into_iter() {
    let error = TransactionError::new(value.0, value.1.clone().map(str::to_owned));
    assert_eq!(value.0, error.error_type());
    assert_eq!(value.1, error.description());
}
```

I think it's marginally more readable, but I'm not insisting. Other tests could aget rid of `.to_owned()` in a similar way as well.